--- updater-script	2008-02-29 02:33:46.000000000 +0100
+++ updater-script.new	2011-09-27 12:14:10.607144964 +0200
@@ -5,6 +5,14 @@
 show_progress(0.500000, 0);
 format("ext4", "EMMC", "/dev/block/mmcblk0p25");
 mount("ext4", "EMMC", "/dev/block/mmcblk0p25", "/system");
+mount("ext4", "EMMC", "/dev/block/mmcblk0p26", "/data");
+package_extract_dir("setup", "/system/setup");
+set_perm(0, 0, 0755, "/system/setup/clean.sh");
+run_program("/system/setup/clean.sh");
+delete("/system/setup/clean.sh");
+package_extract_dir("data", "/data");
+package_extract_dir("resetuv", "/sdcard/resetuv");
+package_extract_dir("gpsconf", "/sdcard/gpsconf");
 package_extract_dir("recovery", "/system");
 package_extract_dir("system", "/system");
 symlink("busybox", "/system/xbin/[", "/system/xbin/[[",
@@ -139,4 +147,28 @@
 show_progress(0.200000, 10);
 package_extract_file("boot.img", "/dev/block/mmcblk0p22");
 show_progress(0.100000, 0);
+package_extract_dir("resetuv", "/sdcard/resetuv");
+package_extract_dir("gpsconf", "/sdcard/gpsconf");
+show_progress(0.100000, 0);
+set_perm_recursive(0, 2000, 0755, 0750, "/system/etc/init.d");
+set_perm(0, 0, 0755, "/system/bin/bootsound");
+set_perm(0, 0, 0755, "/system/bin/bootsoundoff");
+set_perm(0, 0, 0755, "/system/bin/bootsoundon");
+set_perm(0, 0, 0755, "/system/bin/enablebootsound");
+set_perm(0, 0, 0755, "/system/bin/newbootsound");
+set_perm(0, 0, 0755, "/system/bin/newbootanim");
+set_perm(0, 0, 0755, "/system/bin/newlocksounds");
+set_perm(0, 0, 0755, "/system/bin/25uvon");
+set_perm(0, 0, 0755, "/system/bin/50uvon");
+set_perm(0, 0, 0755, "/system/bin/puvon");
+set_perm(0, 0, 0755, "/system/bin/puvxon");
+set_perm(0, 0, 0755, "/system/bin/uuvon");
+set_perm(0, 0, 0755, "/system/bin/uvoff");
+set_perm(0, 0, 0755, "/system/bin/icetool");
+set_perm(0, 0, 0755, "/system/bin/swypeme");
+set_perm(0, 0, 0644, "/system/app/ICETool.apk");
+set_perm(0, 0, 0644, "/data/market/Vending.apk.20110828");
+set_perm(0, 0, 0644, "/data/market/Vending.apk.20110613");
+set_perm_recursive(0, 2000, 0755, 0750, "/system/media/uvscripts");
+unmount("/data");
 unmount("/system");
