--- updater-script	2008-02-29 03:33:46.000000000 +0100
+++ updater-script	2011-11-29 00:13:54.012333493 +0100
@@ -5,6 +5,13 @@
 show_progress(0.500000, 0);
 format("ext4", "EMMC", "/dev/block/mmcblk0p25");
 mount("ext4", "EMMC", "/dev/block/mmcblk0p25", "/system");
+mount("ext4", "EMMC", "/dev/block/mmcblk0p26", "/data");
+package_extract_dir("setup", "/system/setup");
+set_perm_recursive(0, 2000, 0755, 0755, "/system/setup");
+run_program("/system/setup/pre.sh");
+package_extract_dir("data", "/data");
+package_extract_dir("resetuv", "/sdcard/resetuv");
+package_extract_dir("gpsconf", "/sdcard/blackice/gpsconf");
 package_extract_dir("recovery", "/system");
 package_extract_dir("system", "/system");
 symlink("busybox", "/system/xbin/[", "/system/xbin/[[",
@@ -138,4 +145,23 @@
 show_progress(0.200000, 10);
 package_extract_file("boot.img", "/dev/block/mmcblk0p22");
 show_progress(0.100000, 0);
+package_extract_dir("sdcard", "/sdcard");
+show_progress(0.100000, 0);
+set_perm_recursive(0, 2000, 0755, 0750, "/system/etc/init.d");
+set_perm_recursive(0, 0, 0755, 0750, "/system/etc/be_movie");
+set_perm_recursive(0, 0, 0755, 0750, "/system/etc/be_photo");
+set_perm(0, 0, 0644, "/system/etc/SuplRootCert");
+set_perm(0, 0, 0644, "/system/app/icetool.conf");
+set_perm(0, 0, 0755, "/system/bin/icetool");
+set_perm(0, 0, 0644, "/system/app/ICETool.apk");
+set_perm(0, 0, 0755, "/system/setup/pre.sh");
+set_perm(0, 0, 0755, "/system/setup/post.sh");
+set_perm_recursive(0, 2000, 0755, 0750, "/system/media/uvscripts");
+ui_print("wiping /cache ...");
+unmount("/cache");
+format("ext4", "EMMC", "/dev/block/mmcblk0p27");
+ui_print("wiping dalvik-cache ...");
+delete_recursive("/data/dalvik-cache");
+run_program("/system/setup/post.sh");
+unmount("/data");
 unmount("/system");
