diff --git packages/SystemUI/src/com/android/systemui/statusbar/CarrierLabel.java packages/SystemUI/src/com/android/systemui/statusbar/CarrierLabel.java
--- packages/SystemUI/src/com/android/systemui/statusbar/CarrierLabel.java 2012-04-08 03:25:53.651671741 +0800
+++ packages/SystemUI/src/com/android/systemui/statusbar/CarrierLabel.java 2012-04-08 03:09:33.675669996 +0800
@@ -197,7 +197,36 @@
                 break;

             case TYPE_CUSTOM:
-                setText(mCarrierLabelCustom);
+                // If the custom carrier label contains any "$x" items then we must
+                // replace those with the proper text.
+                //  - $n = new line
+                //  - $d = default carrier text
+                //  - $p = plmn carrier text
+                //  - $s = spn carrier text
+                //
+
+                // First we create the default carrier text in case we need it.
+                StringBuilder defaultStr = new StringBuilder();
+                if (showPlmn) {
+                    if (plmn != null) {
+                        defaultStr.append(plmn);
+                    } else {
+                        defaultStr.append(mContext.getText(R.string.lockscreen_carrier_default));
+                    }
+                }
+                if (showSpn && spn != null) {
+                    if (showPlmn) {
+                        defaultStr.append('\n');
+                    }
+                    defaultStr.append(spn);
+                }
+
+                String customStr = mCarrierLabelCustom;
+                customStr = customStr.replaceAll("\\$n", "\n");
+                customStr = customStr.replaceAll("\\$d", (defaultStr != null) ? defaultStr.toString() : "");
+                customStr = customStr.replaceAll("\\$p", (plmn != null) ? plmn : "");
+                customStr = customStr.replaceAll("\\$s", (spn != null) ? spn : "");
+                setText(customStr);
                 break;
         }
     }
diff --git policy/src/com/android/internal/policy/impl/LockScreen.java policy/src/com/android/internal/policy/impl/LockScreen.java
--- policy/src/com/android/internal/policy/impl/LockScreen.java	2012-04-08 17:40:04.039763045 +0800
+++ policy/src/com/android/internal/policy/impl/LockScreen.java	2012-04-08 03:09:52.599670030 +0800
@@ -1445,7 +1424,30 @@
                 }
                 break;
             case CARRIER_TYPE_CUSTOM:
-                return carrierLabelCustom;
+                // If the custom carrier label contains any "$x" items then we must
+                // replace those with the proper text.
+                //  - $n = new line
+                //  - $d = default carrier text
+                //  - $p = plmn carrier text
+                //  - $s = spn carrier text
+                //
+
+                // First we create the default carrier text in case we need it.
+                StringBuilder defaultStr = new StringBuilder();
+                if (telephonyPlmn != null && TextUtils.isEmpty(telephonySpn)) {
+                    defaultStr.append(telephonyPlmn.toString());
+                } else if (telephonySpn != null && TextUtils.isEmpty(telephonyPlmn)) {
+                    defaultStr.append(telephonySpn.toString());
+                } else if (telephonyPlmn != null && telephonySpn != null) {
+                    defaultStr.append(telephonyPlmn.toString() + "|" + telephonySpn.toString());
+                }
+
+                String customStr = carrierLabelCustom;
+                customStr = customStr.replaceAll("\\$n", "\n");
+                customStr = customStr.replaceAll("\\$d", (defaultStr != null) ? defaultStr.toString() : "");
+                customStr = customStr.replaceAll("\\$p", (telephonyPlmn != null) ? telephonyPlmn.toString() : "");
+                customStr = customStr.replaceAll("\\$s", (telephonySpn != null) ? telephonySpn.toString() : "");
+                return customStr;
          }
          return "";
      }
