diff --git policy/src/com/android/internal/policy/impl/LockScreen.java policy/src/com/android/internal/policy/impl/LockScreen.java
--- policy/src/com/android/internal/policy/impl/LockScreen.java	2012-01-26 14:58:14.122573820 +0800
+++ policy/src/com/android/internal/policy/impl/LockScreen.java	2012-01-26 15:12:43.362584406 +0800
@@ -96,11 +96,11 @@
     private static final String TAG = "LockScreen";
     private static final String ENABLE_MENU_KEY_FILE = "/data/local/enable_menu_key";
     private static final Uri sArtworkUri = Uri.parse("content://media/external/audio/albumart");
+    private static final String TOGGLE_FLASHLIGHT = "net.cactii.flash2.TOGGLE_FLASHLIGHT";
     static final int CARRIER_TYPE_DEFAULT = 0;
     static final int CARRIER_TYPE_SPN = 1;
     static final int CARRIER_TYPE_PLMN = 2;
     static final int CARRIER_TYPE_CUSTOM = 3;
-
     private Status mStatus = Status.Normal;

     private LockPatternUtils mLockPatternUtils;
@@ -893,6 +893,14 @@
         updateStatusLines();
     }

+    private void toggleFlashLight() {
+        Intent intent = new Intent(LockScreen.TOGGLE_FLASHLIGHT);
+        intent.putExtra("strobe", false);
+        intent.putExtra("period", 0);
+        intent.putExtra("bright", false);
+        getContext().sendBroadcast(intent);
+    }
+
     @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) {
         if ((keyCode == KeyEvent.KEYCODE_DPAD_CENTER && mTrackballUnlockScreen)
@@ -900,6 +908,19 @@
                 || (keyCode == KeyEvent.KEYCODE_MENU && mEnableMenuKeyInLockScreen)) {

             mCallback.goToUnlockScreen();
+            return false;
+
+        } else if (keyCode == KeyEvent.KEYCODE_HOME) {
+            event.startTracking();
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean onKeyLongPress(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_HOME) {
+          toggleFlashLight();
         }
         return false;
     }
